### PR TITLE
docs(subscriptions): Add starter Subscriptions docs

### PR DESCRIPTION
**Pull request by ayirr7**
When first working on the Create a Dataset onboarding task, I felt that I was unclear on how to build a subscription using Snuba API, and what each field of the subscription payload meant. This is an initial step towards explaining the components of a subscription in more detail. Ideas for further elaboration would be much appreciated!

### CODE REVIEW

**Review by evanh — APPROVED** [review]

**Review by onewland — COMMENTED** [review]
@lynnagara does this seem accurate to you? Is there some more in-depth documentation it should potentially point to?

File: snuba/subscriptions/data.py
Comment (lynnagara) [reply snuba/subscriptions/data.py:87]: Would this docstring be better suited on the create subscription endpoint or in usage documentation instead of on this class? If it aims to help users to create and use subscriptions more easily I think this is buried deep inside the subscriptions implementation.

**Review by lynnagara — CHANGES_REQUESTED** [review]
Content seems good to me, marking request changes here so we can move it to a more obvious spot before merging.

File: snuba/subscriptions/data.py
Comment (ayirr7) [reply snuba/subscriptions/data.py:87]: This is a good point. I'll move it

File: snuba/subscriptions/data.py
Comment (onewland) [inline snuba/subscriptions/data.py:80]:
```suggestion
    subscription query will look at the past 'time_window_sec' seconds (as specified by the
```

**Comment by ayirr7** [comment]
I'm planning on moving this into the Snuba docs in a new section on Subscriptions

File: docs/source/query/overview.rst
Comment (onewland) [inline docs/source/query/overview.rst:275]: I would call this Creating rather than Sending, probably

**Review by onewland — APPROVED** [review]
I don't think we need to block the merge on this feedback but I think it would probably be helpful to explain what actually happens as a result of a successful creation:
1. queries get run roughly every ~resolution
2. query results get sent to the results Kafka topic (usually consumed in Sentry)

**Comment by ayirr7** [comment]
> 1. queries get run roughly every ~resolution
> 2. query results get sent to the results Kafka topic (usually consumed in Sentry)

Sounds good to me, I think I mentioned (1) already but it would be probably be helpful to elaborate. Indeed, my next steps would be to discuss the executor, commit log and results topic, which includes (2).

**Review by lynnagara — APPROVED** [review]
Thanks!

**Pull request merged**
